improve: batched trackers update

jprzimba opened the pull request:
## Tracker Functions Optimization

Batching has been implemented for tracker functions to reduce I/O and allocations:

### **Changes Implemented:**

1. **Batching Data Structure:**
- `BatchedTrackerData` to accumulate loot, supply, impact, and input data
- Control flags to avoid multiple schedules

2. **Optimized Functions:**
- `sendLootStats`: Accumulates items and values, schedules flush in 250ms
- `updateSupplyTracker`: Accumulates items and values, schedules flush in 250ms
- `updateImpactTracker`: Accumulates combat data, schedules flush in 250ms
- `updateInputAnalyzer`: Accumulates input data, schedules flush in 250ms

3. **Flush Function:**
- `flushBatchedTrackerData`: Sends aggregated data and clears the buffer
- Sends aggregated metrics (reduces I/O)
- Sends individual data to client/party (keeps functionality)

### **Benefits:**
- **I/O Reduction**: Metrics sent in batches instead of per event
- **Fewer Allocations**: Data accumulated in reusable structures
- **Coalescence**: Multiple events in the same tick are processed together
- **Controlled Latency**: Flush in 250ms maintains responsiveness

### **Impact:**
- Significantly reduces I/O overhead in high-activity situations (intense combat, mass loot)
- Keeps original functionality intact
- Improves overall server performance under load

jprzimba committed:
fix: batched trackers updated

github-actions[bot] committed:
Code format - (Clang-format)

gabrielew commented on src/creatures/players/player.cpp:
```diff
@@ -1679,47 +1683,128 @@ void Player::sendLootStats(const std::shared_ptr<Item> &item, uint8_t count) {
+				self->flushBatchedTrackerData();
+			},
+			"Player::flushBatchedTrackerData"
+		);
```
if you sum the amount, it'll  create a bug with the "max-dps" and "all-time high"
But if you don’t sum it, you’ll send all of them separately so it’ll be the same, right?

jprzimba committed:
fix: duplicated data in loot metrics

- Previous behavior: flushBatchedTrackerData emitted a single aggregated counter using batchedTrackerData.lootValue ( src/creatures/players/player.cpp:1756 ).
- Change: Emit one counter per loot item using its computed value (coins or “Lootmonger” sell price) during the loop over lootItems . Removed the aggregated emission.
- Result: Metrics reflect each loot event individually and avoid double-reporting caused by batch aggregation.

jprzimba committed:
Reduce tracker batch delay and simplify loot stats

Decreased the tracker batch event delay from 250ms to 50ms for faster updates. Simplified the loot value calculation in flushBatchedTrackerData by aggregating loot value directly and sending metrics only if lootValue is positive. Also added calls to sendStats() and sendSkills() in onCreatureAppear.
